homeassistant/components/fibaro/sensor.py

```diff
@@ -72,44 +76,31 @@ def __init__(self, fibaro_device):
         super().__init__(fibaro_device)
         self.entity_id = f"{DOMAIN}.{self.ha_id}"
         if fibaro_device.type in SENSOR_TYPES:
-            self._unit = SENSOR_TYPES[fibaro_device.type][1]
-            self._icon = SENSOR_TYPES[fibaro_device.type][2]
-            self._device_class = SENSOR_TYPES[fibaro_device.type][3]
+            self.entity_description = SENSOR_TYPES[fibaro_device.type]
+            self._attr_native_unit_of_measurement = (
+                self.entity_description.native_unit_of_measurement
```

Comment:
You don't need to do this; `SensorEntity` will pull it from the description.

Reply:
_attr_native_unit_of_measurement takes precedence over the native_unit_of_measurement of the entity_description. As the entity_description should be constant according to the documentation and in fibaro sensor the same entity_description is used for different units, we need to use former so that it can be configured per sensor instance independent of the entity_description. Unfortunately I don't know all possible combinations of class and unit so I would leave this pattern as is.

Reply:
> _attr_native_unit_of_measurement takes precedence over the native_unit_of_measurement of the entity_description.

Yes, exactly, which is why you don't need to set it here. The override logic below only occurs if `_attr_native_unit_of_measurement` is `None`, so it's equivalent to just check it on the entity description directly. To get concrete, what i'm suggesting is https://github.com/jkeljo/home-assistant/commit/3350917a4fc872de6d90a361b91c8cd8c0fdff84

Reply:
Thanks for the hint, I changed it accordingly.